tests/c/stats4.c

```diff
@@ -0,0 +1,35 @@
+// Run-time:
+//   env-var: YKD_STATS=-
+//   stderr:
+//     {
+//       ...
+//       "traces_compiled_ok": ...
+//       ...
+//     }
+
+// This tests that spun-up compile workers still cause stats to be printed out
+// (i.e. that, because of reference counting, they don't keep `Mt` alive
+// forever).
+
+#include <assert.h>
+#include <setjmp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
```

Comment:
Would a threshold of 8 maximise the possibility that the main thread exits before a compilation thread?

Reply:
I'm not sure I understand the question, but there will always be scenarios where something can kill the system (e.g. with `exit` on the main thread) in a way that won't cause stats to be printed. This PR fixes one of those scenarios, but we probably can't fix all of them.

Reply:
Hrm, I think we crossed wires.

I think the intention of this test is to show that if a compilation thread is still running when the main thread dies, that stats are still printed.

If that's the case, then a threshold of 0 gives the compilation thread 9 iterations worth of time to run before the main thread exits.

But if you set it to 8, you give it 0 iterations of time to run before the main thread exits.

The latter is more likely to capture what you want to test, isn't it? In the former you give the compilation thread more time to finish, and it may exit before the main thread.

Reply:
Understood, but we want to eventually have a fast code generator. If it works with 8, then let's maximise the chances that we test the property we want to, and use 8?

Maybe a more direct approach is to use a loop with only 2 iterations and set the threshold to 1?

Reply:
I think we're over thinking this tiny change, and trying to predict the future when we can just wait and see.

Reply:
Just to be clear: this test fails without the associated change. So the test is testing *something* meaningful.

Reply:
I'm not certain you've understood my explanations.

Do you agree with the following statements?

 - Which thread exits first is non-deterministic.
 - You want to test the case where the main thread exits first.
 - The more time the main thread runs (the remained of the loop iterations after the compilation thread has started), the lower the probability that the main thread will exit first.

If yes, to all of the above, I think you can make a more robust test by simply lowering the loop bound to `hot_threshold + 1`.

Reply:
Changed in c4aa14d.